sdk/python/kfp/dsl/types.py

```diff
@@ -17,187 +17,215 @@
+    else:
+      warnings.warn(InconsistentTypeWarning(error_text))
+  return types_are_compatible
 
 
 def check_types(checked_type, expected_type):
-	'''check_types checks the type consistency.
-	For each of the attribute in checked_type, there is the same attribute in expected_type with the same value.
-	However, expected_type could contain more attributes that checked_type does not contain.
-	Args:
-		checked_type (BaseType/str/dict): it describes a type from the upstream component output
-		expected_type (BaseType/str/dict): it describes a type from the downstream component input
-		'''
-	if isinstance(checked_type, BaseType):
-		checked_type = _instance_to_dict(checked_type)
-	elif isinstance(checked_type, str):
-		checked_type = {checked_type: {}}
-	if isinstance(expected_type, BaseType):
-		expected_type = _instance_to_dict(expected_type)
-	elif isinstance(expected_type, str):
-		expected_type = {expected_type: {}}
-	return _check_dict_types(checked_type, expected_type)
+  """check_types checks the type consistency.
+
+  For each of the attribute in checked_type, there is the same attribute in expected_type with the same value.
+  However, expected_type could contain more attributes that checked_type does not contain.
+  Args:
+          checked_type (BaseType/str/dict): it describes a type from the upstream component output
+          expected_type (BaseType/str/dict): it describes a type from the downstream component input
+
+  """
+  if isinstance(checked_type, BaseType):
+    checked_type = _instance_to_dict(checked_type)
+  elif isinstance(checked_type, str):
+    checked_type = {checked_type: {}}
+  if isinstance(expected_type, BaseType):
+    expected_type = _instance_to_dict(expected_type)
+  elif isinstance(expected_type, str):
+    expected_type = {expected_type: {}}
+  return _check_dict_types(checked_type, expected_type)
+
 
 def _check_valid_type_dict(payload):
-	'''_check_valid_type_dict checks whether a dict is a correct serialization of a type
+  '''_check_valid_type_dict checks whether a dict is a correct serialization of a type
 	Args:
 		payload(dict)
 	'''
-	if not isinstance(payload, dict) or len(payload) != 1:
-		return False
-	for type_name in payload:
-		if not isinstance(payload[type_name], dict):
-			return False
-		property_types = (int, str, float, bool)
-		property_value_types = (int, str, float, bool, dict)
-		for property_name in payload[type_name]:
-			if not isinstance(property_name, property_types) or not isinstance(payload[type_name][property_name], property_value_types):
-				return False
-	return True
+  if not isinstance(payload, dict) or len(payload) != 1:
+    return False
+  for type_name in payload:
+    if not isinstance(payload[type_name], dict):
+      return False
+    property_types = (int, str, float, bool)
+    property_value_types = (int, str, float, bool, dict)
+    for property_name in payload[type_name]:
+      if not isinstance(property_name, property_types) or not isinstance(
```

Comment:
Maybe we should wrap before `or` here.

Reply:
I guess the reason was that it tries to fix 80 chars line limit.